aws/resource_aws_ec2_managed_prefix_list_test.go

```diff
@@ -0,0 +1,750 @@
+			return fmt.Errorf("expected entries %#v, got %#v", expectEntries, entries)
+		}
+
+		return nil
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsEc2ManagedPrefixListDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:       testAccAwsEc2ManagedPrefixListConfig_basic_create,
+				ResourceName: resourceName,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccAwsEc2ManagedPrefixListExists(resourceName, &pl, &entries),
+					checkAttributes,
+					resource.TestCheckResourceAttr(resourceName, "name", "tf-test-basic-create"),
+					testAccMatchResourceAttrRegionalARN(resourceName, "arn", "ec2", regexp.MustCompile(`prefix-list/pl-[[:xdigit:]]+`)),
+					resource.TestCheckResourceAttr(resourceName, "address_family", "IPv4"),
+					resource.TestCheckResourceAttr(resourceName, "max_entries", "5"),
+					resource.TestCheckResourceAttr(resourceName, "entry.#", "2"),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "entry.*", map[string]string{
+						"cidr_block":  "1.0.0.0/8",
+						"description": "Test1",
+					}),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "entry.*", map[string]string{
+						"cidr_block":  "2.0.0.0/8",
+						"description": "Test2",
+					}),
+					testAccCheckResourceAttrAccountID(resourceName, "owner_id"),
+					testAccCheckAwsEc2ManagedPrefixListVersion(&pl, 1),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Key1", "Value1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Key2", "Value2"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config:       testAccAwsEc2ManagedPrefixListConfig_basic_update,
+				ResourceName: resourceName,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccAwsEc2ManagedPrefixListExists(resourceName, &pl, &entries),
+					resource.TestCheckResourceAttr(resourceName, "name", "tf-test-basic-update"),
+					resource.TestCheckResourceAttr(resourceName, "entry.#", "2"),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "entry.*", map[string]string{
+						"cidr_block":  "1.0.0.0/8",
+						"description": "Test1",
+					}),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "entry.*", map[string]string{
+						"cidr_block":  "3.0.0.0/8",
+						"description": "Test3",
+					}),
+					testAccCheckAwsEc2ManagedPrefixListVersion(&pl, 2),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Key1", "Value1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Key3", "Value3"),
+				),
+			},
+		},
+	})
+}
+
+const testAccAwsEc2ManagedPrefixListConfig_basic_create = `
+resource "aws_ec2_managed_prefix_list" "test" {
+  name           = "tf-test-basic-create"
```

Comment:
We should randomize the naming of these resources -- please see the [Contributing Guide](https://github.com/hashicorp/terraform-provider-aws/blob/master/docs/contributing/running-and-writing-acceptance-tests.md#randomized-naming) for information about this (or searching for `rName` usage elsewhere).